File: objective-c/difference-of-squares/DifferenceOfSquares.m
#import "DifferenceOfSquares.h"

@implementation DifferenceOfSquares

static int _differenceOfSquares; + (int)differenceOfSquares {
    return _differenceOfSquares;
} + (void)setDifferenceOfSquares:(int)newDifferenceOfSquares {
    _differenceOfSquares = differenceOfSquares;
}

static int _squareOfSum; + (int)squareOfSum {
    return _squareOfSum;
} + (void)setSquareOfSum:(int)newSquareOfSum {
    _squareOfSum = newSquareOfSum;
}

static int _sumOfSquares; + (int)sumOfSquares {
    return _sumOfSquares;
} + (void)setSumOfSquares:(int)newSumOfSquares {
    _sumOfSquares = newSumOfSquares;
}

- (DifferenceOfSquares *)initWithMax:(int)n {

    int i;
    double doubleRes1 = 0;
    double doubleRes2 = 0;

    for (i = 1; i <= n; i++) {
        doubleRes1 += i;
        doubleRes2 += pow(i, 2);
    }

    _squareOfSum = (int)pow(doubleRes1, 2);
    _sumOfSquares = (int)doubleRes2;
    _differenceOfSquares = abs(_squareOfSum - _sumOfSquares);

    return self;
}


@end

File: cpp/difference-of-squares/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares
{
    // 합의 제곱을 계산
    long square_of_sum(int n)
    {
        long sum = n * (n + 1) / 2; // 등차수열의 합 공식
        return sum * sum;
    }

    // 제곱의 합을 계산
    long sum_of_squares(int n)
    {
        return n * (n + 1) * (2 * n + 1) / 6; // 제곱의 합 공식
    }

    // 두 값의 차이를 계산
    long difference(int n)
    {
        return square_of_sum(n) - sum_of_squares(n);
    }
} // namespace difference_of_squares
